templates/cmd/main.go.tpl

```diff
@@ -72,6 +72,9 @@ type dependencies struct{
 func main() { //nolint: funlen // Why: We can't dwindle this down anymore without adding complexity.
   exitCode := 1
 	defer func() {
+		if r := recover(); r != nil {
+			fmt.Fprintf(os.Stderr, "Recovered from panic: %+v\n%s\n", r, string(debug.Stack()))
+		}
```

Comment:
Sigh. I always forget that a `defer` will just swallow a panic. Good catch, thank you!

This seems like it's a bit lossy compared to the original panic:

**This**: https://go.dev/play/p/JGlQ4lDZv5t?v=goprev
**Original**: https://go.dev/play/p/Qi2xCxybf8R?v=goprev

I suggest we just panic again ([go playground [link]](https://go.dev/play/p/vDS_BRsWoHz?v=goprev)):

```suggestion
		if r := recover(); r != nil {
			panic(r)
		}
```

This is generally what is done when you want to re-throw a panic. It introduces another layer, but I think it's better to not hide the fact that we're handling and re-throwing it.